### PR TITLE
Show lock owner instead of repo owner on LFS setting page

### LINKED ISSUE #31784: LFS Lock list should display lock owner instead of repository owner
### Description

`$.Owner` here seems to be repository owner instead of LFS Lock owner
https://github.com/go-gitea/gitea/blob/94cca8846e7d62c8a295d70c8199d706dfa60e5c/templates/repo/settings/lfs_locks.tmpl#L33-L36

can not reproduce on demo site, because lfs not enabled

### Screenshots

image

### Gitea Version

1.22.1

### Can you reproduce the bug on the Gitea demo site?

No

### Operating System

win11 22H2

### Browser Version

chrome 127

### PR DESCRIPTION
**Pull request by wolfogre**
Fix #31784.

Before:

image

After:

image

### CODE REVIEW

File: templates/repo/settings/lfs_locks.tmpl [hunk 1]
```diff
@@ -30,9 +30,9 @@
 								{{end}}
 							</td>
 							<td>
-								<a href="{{$.Owner.HomeLink}}">
-									{{ctx.AvatarUtils.Avatar $.Owner}}
-									{{$.Owner.DisplayName}}
+								<a href="{{.Owner.HomeLink}}">
```
Comment (lunny) [inline templates/repo/settings/lfs_locks.tmpl]: `$lock` ?

**Review by lunny — APPROVED** [review]

File: templates/repo/settings/lfs_locks.tmpl (hunk 1 above)
Comment (wolfogre) [reply templates/repo/settings/lfs_locks.tmpl]: I think it's the same, like `.Created`.

image

But maybe `$lock` can make it clearer.

File: models/git/lfs_lock_list.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+	})
+	users := make(map[int64]*user_model.User, len(usersIDs))
+	if err := db.GetEngine(ctx).
+		In("id", usersIDs).
+		Find(&users); err != nil {
+		return fmt.Errorf("find users: %w", err)
+	}
+	for _, v := range locks {
+		v.Owner = users[v.OwnerID]
+	}
```
Comment (wxiaoguang) [inline models/git/lfs_lock_list.go:51]: It will cause nil panic in template

File: models/git/lfs_lock.go [hunk 3]
```diff
@@ -37,6 +39,31 @@ func (l *LFSLock) BeforeInsert() {
...
+func (l *LFSLock) LoadOwner(ctx context.Context) error {
+	if l.Owner != nil {
+		return nil
+	}
+
+	owner, err := user_model.GetUserByID(ctx, l.OwnerID)
+	if err != nil {
+		return err
+	}
+	l.Owner = owner
```
Comment (wxiaoguang) [inline models/git/lfs_lock.go:66]: It will cause unnecessary 500 error when a user doesn't exist.

File: models/git/lfs_lock_list.go (hunk 2 above)
Comment (wolfogre) [reply models/git/lfs_lock_list.go:51]: 765ff4bb547fa4764bd75cb10844e84c5bb91794

File: models/git/lfs_lock.go (hunk 3 above)
Comment (wolfogre) [reply models/git/lfs_lock.go:66]: 765ff4bb547fa4764bd75cb10844e84c5bb91794

**Review by lafriks — APPROVED** [review]

**Pull request merged**